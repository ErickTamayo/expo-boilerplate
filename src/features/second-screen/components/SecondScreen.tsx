import * as React from 'react'
import { StyleSheet, Text, View, Button } from 'react-native'
import { NavigationInjectedProps } from 'react-navigation'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
})

export interface Props extends NavigationInjectedProps {}

export default class SecondScreen extends React.Component<Props, object> {
  public render(): JSX.Element {
    const { navigation } = this.props

    return (
      <View style={styles.container}>
        <Text>Second Screen</Text>
        <Button onPress={() => navigation.navigate('Start')} title="Go Back" color="#841584" />
      </View>
    )
  }
}
