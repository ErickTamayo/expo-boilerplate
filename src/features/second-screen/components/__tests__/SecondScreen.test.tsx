import 'react-native'
import * as React from 'react'
import { shallow } from 'enzyme'
import SecondScreen, { Props } from '../SecondScreen'
import { NavigationScreenProp, NavigationRoute, NavigationParams } from 'react-navigation'

describe('<SecondScreen />', () => {
  it('renders correctly', () => {
    const props: Props = {
      navigation: {} as NavigationScreenProp<NavigationRoute<NavigationParams>, NavigationParams>,
    }

    const wrapper = shallow(<SecondScreen {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
