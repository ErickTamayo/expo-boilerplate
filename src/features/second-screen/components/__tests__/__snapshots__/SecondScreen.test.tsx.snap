// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecondScreen /> renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component>
    Second Screen
  </Component>
  <Button
    color="#841584"
    onPress={[Function]}
    title="Go Back"
  />
</Component>
`;
